core/trino-main/src/main/java/io/trino/metadata/FunctionMetadata.java

```diff
@@ -85,7 +116,7 @@ public FunctionId getFunctionId()
 
     public Signature getSignature()
     {
-        return signature;
+        return canonicalSignature;
```

Comment:
If you do not rename the getter (which I am OK with), please keep the field name as it was.
Having different field and getter names creates confusion.

Reply:
Indeed, this creates confusion. Rename methods getSignature will affect too many places, so I'll revert the naming.